src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -44,14 +46,18 @@ private IOUtilities() {}
      * @param prefix     The prefix of temporary files to clean up, or null if all
      *                   temporary files should be removed.
      *
-     * @throws IOException if an I/O error occurs during deletion.
+     * @implNote IOException is wrapped to RuntimeException, which will be logged to the error log by the ioPool.
      */
-    public static void cleanupTempFiles(Path targetPath, @Nullable String prefix) throws IOException {
-        try (var filesToDelete = Files.find(targetPath, 1, createPredicate(prefix))) {
-            for (var file : filesToDelete.toList()) {
-                Files.deleteIfExists(file);
+    public static CompletableFuture<?> cleanupTempFiles(CompletableFuture<?> pendingWriteFuture, Path targetPath, @Nullable String prefix) {
+        return pendingWriteFuture.thenRunAsync(() -> {
+            try (var filesToDelete = Files.find(targetPath, 1, createPredicate(prefix))) {
+                for (var file : filesToDelete.toList()) {
+                    Files.deleteIfExists(file);
+                }
```

Comment:
As a general principle, this will also throw an exception if it fails. Should we still attempt to cleanup other temporary files even if one throws an I/O exception? I believe we should.

Reply:
based on how the temp file is created, there should only ever be one match.

This PR is about moving the existing code to a thread, not really revisiting existing Neo code...

Reply:
In my opinion, that's not a reason to change behavior, but if it should only match one file per operation, it's fine.